src/core/server/saved_objects/import/regenerate_ids.ts

```diff
@@ -42,3 +43,40 @@ export const regenerateIds = (objects: SavedObject[]) => {
   }, new Map<string, { id: string; omitOriginId?: boolean }>());
   return importIdMap;
 };
+
+export const regenerateIdsWithReference = async (props: {
```

Comment:
Do we need unit tests for `regenerateIdsWithReference`?

Reply:
Sure, let me do that.